bee-plugin/src/handler.rs

```diff
@@ -0,0 +1,218 @@
+// Copyright 2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    error::PluginError,
+    event::EventId,
+    grpc::{plugin_client::PluginClient, DummyEvent, ShutdownRequest, SillyEvent},
+    handshake::HandshakeInfo,
+    streamer::PluginStreamer,
+    PluginId, UniqueId,
+};
+
+use bee_event_bus::EventBus;
+
+use tokio::{
+    io::{AsyncBufReadExt, BufReader},
+    process::{Child, Command},
+    select, spawn,
+    sync::{mpsc::unbounded_channel, oneshot::Sender},
+    task::JoinHandle,
+    time::sleep,
+};
+use tonic::{transport::Channel, Request};
+
+use std::{
+    any::type_name,
+    collections::{hash_map::Entry, HashMap},
+    process::Stdio,
+    time::Duration,
+};
+
+macro_rules! spawn_streamers {
```

Comment:
I wonder if that macro would better live in the `streamer` module? (I think from a user pov one would expect it to be there idk)

Reply:
Well sort of because the macro is actually called in this module and this module and all its types are private, so the user doesn't even see the `PluginHandler` values.